### DISCUSSION TITLE
_add_holiday* method documentation

### DISCUSSION THREAD

**Discussion by maxrausch** [vacanza/holidays#1665] [Q&A] [answered]
Is there any documentation page that explains the _add_holiday method where you add a specific day in each. I was looking through the code and the logic seems to be based on a regex of the name and ends up calling _get_nth_weekday_from eventually but there are no examples in the document that explain how to call these functions for creating a new customized exchange holiday calendar.

**Comment by arkid15r** [comment, marked as answer]
Hi @maxrausch, 
you're absolutely right -- there is no documentation for these methods. You can blame me for that. I implemented those methods as a part of syntactic sugar improvements for PH needs. You can still use the "old fashioned" way for adding holidays directly, e.g. `entity[dt] = "My Holiday"`. [Here](https://github.com/vacanza/python-holidays/blob/1845ce1fc2e2cb52f402db5a66fd11a068d15af1/holidays/holiday_base.py#L428) you can find the implementation with my comments. Essentially the code supports 4 different cases (from most to least frequently used):
  -  `<month> <day>` patterns (e.g., _add_holiday_jun_15()).
  -  `<last/nth> <weekday> of <month>` patterns (e.g., _add_holiday_last_mon_of_aug() or _add_holiday_3rd_fri_of_aug()).
  - `<n> day(s) <past/prior> <last/<nth> <weekday> of <month>` patterns (e.g., _add_holiday_1_day_past_1st_fri_of_aug() or _add_holiday_5_days_prior_last_fri_of_aug()).
  - `<nth> <weekday> <before/from> <month> <day>` patterns (e.g.,  _add_holiday_1st_mon_before_jun_15() or _add_holiday_1st_mon_from_jun_15()).

I'll create an issue for documenting this.

**Reply by maxrausch** [reply]
Thank you! This is very helpful. I don't think I understood the "old fashion" way but I re-read this section and makes sense now: https://python-holidays.readthedocs.io/en/latest/examples.html#creating-custom-holidays-or-augmenting-existing-ones-with-private-ones

**Comment by Kartheekmurala** [comment]
Hi @arkid15r @KJhellico @PPsyrius ! I’m preparing my GSoC 2026 proposal for the holidays project.

I’ve drafted my approach and would really appreciate any high-level feedback—especially on whether the scope and timeline seem reasonable.

Here’s my draft:
https://docs.google.com/document/d/1GJsWEhrJ-V7hL2EQjHb9iOpd7chY6RWw0xTKYXGHEW8/edit?tab=t.0

If possible, I’d also love suggestions on improving alignment with the project goals.

I understand mentors are busy, so even brief feedback would be very helpful. Thank you!

**Comment by CodeWithHarsh143** [comment]
hi I'm new to open source and I'm very interested in contributing to open world  **python-holidays**. Over the past week, I've been exploring the project, reading through the documentation, and using the library to better understand how it works from a user's perspective.

I'm now looking to make my first contribution. If there are any beginner-friendly issues or tasks that would be a good starting point, I'd really appreciate your guidance or suggestions.

Thank you for maintaining this project, and I look forward to contributing!
